frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/index.tsx

```diff
@@ -90,42 +90,58 @@ const BasicSelector: React.SFC<SpecCapabilityProps> = ({ value, capability }) =>
 );
 
 const BooleanSwitch: React.FC<SpecCapabilityProps> = (props) => {
+  const { onHandleError } = props;
   const convertedValue = !_.isNil(props.value) ? props.value : false;
   const [value, setValue] = React.useState(convertedValue);
   const [confirmed, setConfirmed] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState(null);
 
-  const patchFor = (val: boolean) => [
-    { op: 'add', path: `/spec/${props.descriptor.path.replace('.', '/')}`, value: val },
-  ];
   const update = () => {
     setConfirmed(true);
-    return k8sPatch(props.model, props.obj, patchFor(value));
+    setErrorMessage(null);
+
+    let newObj = _.cloneDeep(props.obj);
+    newObj = _.set(newObj, `spec.${props.descriptor.path}`, value);
+
+    return k8sUpdate(props.model, newObj).catch((e) => {
```

Comment:
We should prefer `k8sPatch` to `k8sUpdate` because it avoids errors if the object was updated in the background by another user.

Reply:
@spadgett  Hmm, I didn't think of that.
I tried to fix the bug with `k8sPatch` method  but it didn't work for unset property. 
During form creation, some of the toggle fields property have no default value set except the user interact with it, so patch update failed. I was thinking of setting those toggle fields property to false during creation, however @TheRealJon raised a concern -"The problem is, if you have a boolean field that's nested in an object or array field, then if the user hasn't set any other value in that object or array, it will still be included in the form data since the boolean value is set". 
Is there a way to use `k8sPatch` to update unset property?

Reply:
Do you mean if the parent object doesn't exist? This gets a little tricky. You would have to add the entire parent object. For arrays, it's similar. See the fix #806.

If it gets too complicated, I'm OK with `k8sUpdate`. I would add a comment explaining why we're not using patch.

Reply:
Maybe we use `k8sPatch` if the parent exists, but fall back to `k8sUpdate` when it doesn't.

Reply:
@spadgett Sounds good. Thanks.